Feature/SDLWIN-142

AGaliuzov opened the pull request:
Add implementation for Qt database wrapper
Related-Issue: [SDLWIN-142](https://adc.luxoft.com/jira/browse/SDLWIN-14://adc.luxoft.com/jira/browse/SDLWIN-142)

AGaliuzov committed:
Remove temporrary added WIN_NATIVE ifdefs

Since new impllementation for QtSql has been writen
this temporary fix can be removed

AGaliuzov committed:
Add implementation for Qt database wrapper

Related-Issue: [SDLWIN-142](https://adc.luxoft.com/jira/browse/SDLWIN-14://adc.luxoft.com/jira/browse/SDLWIN-142)

AGaliuzov commented:
guys just review do not merge, I want to check non QT version before

AGaliuzov committed:
Fix build of sql stuf for non Qt version

AGaliuzov commented:
Pull request has been updated and now can be merged

anosach-luxoft commented:
reviewed

pestOO commented on src/components/utils/include/utils/sql_qt_wrapper/sql_database.h:
```diff
@@ -0,0 +1,133 @@
+
+  operator QSqlDatabase() const;
+
+ protected:
```
redundnat

pestOO commented on src/components/utils/include/utils/sql_qt_wrapper/sql_database.h:
```diff
@@ -0,0 +1,133 @@
+  operator QSqlDatabase() const;
+
+ protected:
+ private:
```
Indent according to gstyle

pestOO commented on src/components/utils/src/sql_qt_wrapper/sql_database.cc:
```diff
@@ -0,0 +1,72 @@
+}
+
+bool SQLDatabase::Exec(const std::string& query) {
+  return true;
```
May be add todo/fixme?

AGaliuzov replied:
for what? it is final implementation)))

pestOO commented on src/components/utils/src/sql_qt_wrapper/sql_query.cc:
```diff
@@ -0,0 +1,159 @@
+}
+
+bool SQLQuery::Prepare(const std::string& query) {
+  return query_.prepare(query.c_str());
```
`c_str` seems redundant

AGaliuzov replied:
Actually `QString` can't be created from `std::string`

pestOO replied:
You are god damn right!
In this case may be use `QString::fromUtf8`?

AGaliuzov replied:
Yes, good suggestion

AGaliuzov replied:
@pestOO , I investigate this question bit deeply and I can see that this `const char` will be converted to `Unicode`, so no neccessary to explicitly call `fromUtf8`

pestOO commented on src/components/utils/src/sql_qt_wrapper/sql_query.cc:
```diff
@@ -0,0 +1,159 @@
+}
+
+double SQLQuery::GetDouble(int pos) const {
+  const QVariant val = query_.value(pos);
```
May de verify and log in case of no valeu with this position?

AGaliuzov replied:
Actuallu all logging on the upper level. I just tried to keep  consistency with other implementations for `SQL`

malirod commented:
@AGaliuzov, qt port should not require sqllite. And current develop requires it, even somewhere in the `prop_Qt_x86.cmd`. I see no changes in and cmd file so I assume it's still required.

Our wiki page related to the setup also requires sqllite as the common requirement. I think this should be fixed too.

One more thing. The link to the sqllite points to the sources only. But we need binaries too. Otherwise the build file generation will fail(`.def` is not found).

AGaliuzov commented:
@malirod I'll update it

AGaliuzov committed:
Add proper logic for `Next()` and `Exec()` methods

Als modify cmake files oin order to avoid unneccessary
libraries usage

Related issue: SDLWIN-142

AGaliuzov commented:
@malirod , see updated PR

AGaliuzov committed:
Fix `HasErrors()` method for other sql wrapper

Related issue SDLWIN-142

AGaliuzov commented:
@malirod , update `HasErrors`

AGaliuzov committed:
Modify comments style. Cahnge wrong descriptions

Related issue SDLWIN-142

AGaliuzov committed:
Substitude MACRO usage with function

Related issue SDLWIN-142

AGaliuzov committed:
Remove constness from sql query getters

since they change query state
Related issue SDLWIN-142

AGaliuzov commented:
@malirod see [158daf8](https://github.com/AGaliuzov/sdl_core_winport/commit/158daf8bdc4c4800332517037fd6ca8ab78a0c12) where constness is removed

AGaliuzov committed:
Move sql_wrapper.h into utils

Move this file in order to allow proper file to be included
during compilation of resumption as well as policy

Related-Issue SDLWIN-142

AGaliuzov commented:
@malirod, please see [here](https://github.com/AGaliuzov/sdl_core_winport/commit/7c053e27e9fba47392717d8edf4177ac079bd24f) for resolving issue with linking
